### PR TITLE
Pm permission to update igv

### PR DESCRIPTION
**Pull request by hanars**
Grants PMs permission to update IGV. Moves the relevent functions out of dataset_api.py into the more pertinent igv_api.py

### CODE REVIEW

File: seqr/views/apis/dataset_api.py [hunk 1]
```diff
@@ -125,110 +121,6 @@ def add_variants_dataset_handler(request, project_guid):
     return create_json_response(response_json)
 
 
-@data_manager_required
```
Comment (hanars) [inline seqr/views/apis/dataset_api.py:128]: all copy-pasted to igv_api.py

File: seqr/views/apis/igv_api.py [hunk 2]
```diff
@@ -1,14 +1,129 @@
...
+from seqr.models import Individual, IgvSample
 from seqr.utils.file_utils import file_iter, does_file_exist
-from seqr.views.utils.permissions_utils import get_project_and_check_permissions
+from seqr.views.utils.file_utils import save_uploaded_file
+from seqr.views.utils.json_to_orm_utils import get_or_create_model_from_json
+from seqr.views.utils.json_utils import create_json_response
+from seqr.views.utils.orm_to_json_utils import  get_json_for_sample
+from seqr.views.utils.permissions_utils import get_project_and_check_permissions, check_project_permissions, \
+    user_is_data_manager, user_is_pm
 from settings import API_LOGIN_REQUIRED_URL
```
Comment (hanars) [inline seqr/views/apis/igv_api.py:16]: unless noted otherwise, everything is copy-paste

File: seqr/views/apis/igv_api.py [hunk 3]
```diff
@@ -1,14 +1,129 @@
...
 
 import logging
 logger = logging.getLogger(__name__)
 
+pm_or_data_manager_required = user_passes_test(
+    lambda user: user_is_data_manager(user) or user_is_pm(user), login_url=API_LOGIN_REQUIRED_URL)
+
+@pm_or_data_manager_required
```
Comment (hanars) [inline seqr/views/apis/igv_api.py:23]: this is the real thing that changed - checks if user is either PM or data manager

File: seqr/views/apis/igv_api.py [hunk 4]
```diff
@@ -1,14 +1,129 @@
...
+
+SAMPLE_TYPE_MAP = {
+    'bam': IgvSample.SAMPLE_TYPE_ALIGNMENT,
+    'cram': IgvSample.SAMPLE_TYPE_ALIGNMENT,
+    'bigWig': IgvSample.SAMPLE_TYPE_COVERAGE,
+    'junctions.bed.gz': IgvSample.SAMPLE_TYPE_JUNCTION,
+    'dcr.bed.gz': IgvSample.SAMPLE_TYPE_GCNV,
+}
+
+@pm_or_data_manager_required
```
Comment (hanars) [inline seqr/views/apis/igv_api.py:89]: changed permission check

File: seqr/views/apis/igv_api.py [hunk 5]
```diff
@@ -1,14 +1,129 @@
...
+        file_path = request_json.get('filePath')
+        if not file_path:
+            raise ValueError('request must contain fields: filePath')
+
+        suffix = '.'.join(file_path.split('.')[1:])
+        sample_type = SAMPLE_TYPE_MAP.get(suffix)
+        if not sample_type:
+            raise Exception('Invalid file extension for "{}" - valid extensions are {}'.format(
+                file_path, ', '.join(SAMPLE_TYPE_MAP.keys())))
+        if not does_file_exist(file_path):
```
Comment (hanars) [inline seqr/views/apis/igv_api.py:107]: changed this to directly check the file here (used to have a helper method but that didn't hep much)

**Review by ShifaSZ — CHANGES_REQUESTED** [review]
There are some CI tests failed.

**Review by ShifaSZ — APPROVED** [review]

**Pull request merged**